clarity/src/vm/docs/mod.rs

```diff
@@ -102,20 +110,23 @@ const TX_SENDER_KEYWORD: KeywordAPI = KeywordAPI {
     description: "Returns the original sender of the current transaction, or if `as-contract` was called to modify the sending context, it returns that
 contract principal.",
     example: "(print tx-sender) ;; Will print out a Stacks address of the transaction sender",
+    version: ClarityVersion::Clarity1,
 };
 
 const TX_SPONSOR_KEYWORD: KeywordAPI = KeywordAPI {
     name: "tx-sponsor?",
     output_type: "optional principal",
     description: "Returns the sponsor of the current transaction if there is a sponsor, otherwise returns None.",
     example: "(print tx-sponsor?) ;; Will print out an optional value containing the Stacks address of the transaction sponsor",
+    version: ClarityVersion::Clarity1,
```

Comment:
`tx-sponsor?` is `Clarity2`.

Rather than re-specifying the version for all the keywords, it'd be better to use the `NativeVariables` field.

I think the way I'd refactor this is to add new internal struct `SimpleKeywordAPI` which was:

```rust
struct SimpleKeywordAPI {
    name: &'static str,
    output_type: &'static str,
    description: &'static str,
    example: &'static str,
}
```

And then change all these consts to be instances of that struct rather than `KeywordAPI`.

Then, `make_keyword_reference` would be something like:

```rust
fn make_keyword_reference(variable: &NativeVariables) -> KeywordAPI {
    let simple_api = match variable {
        NativeVariables::TxSender => TX_SENDER_KEYWORD.clone(),
        NativeVariables::ContractCaller => CONTRACT_CALLER_KEYWORD.clone(),
        NativeVariables::NativeNone => NONE_KEYWORD.clone(),
        NativeVariables::NativeTrue => TRUE_KEYWORD.clone(),
        NativeVariables::NativeFalse => FALSE_KEYWORD.clone(),
        NativeVariables::BlockHeight => BLOCK_HEIGHT.clone(),
        NativeVariables::BurnBlockHeight => BURN_BLOCK_HEIGHT.clone(),
        NativeVariables::TotalLiquidMicroSTX => TOTAL_LIQUID_USTX_KEYWORD.clone(),
        NativeVariables::Regtest => REGTEST_KEYWORD.clone(),
        NativeVariables::Mainnet => MAINNET_KEYWORD.clone(),
        NativeVariables::ChainId => CHAINID_KEYWORD.clone(),
        NativeVariables::TxSponsor => TX_SPONSOR_KEYWORD.clone(),
    };
    KeywordAPI {
        // name could even be `variable.get_name()` here, and then
        // `SimpleKeywordAPI` could drop the name field
        name: simple_api.name,
        output_type: simple_api.output_type,
        description: simple_api.description,
        example: simple_api.example,
        version: variable.get_version(),
    }
}
```

That way, we don't end up in a situation where versions need to be specified in two different places.

Reply:
did this